shared/chat/conversation/list-area/normal/index.native.js

```diff
@@ -60,20 +87,103 @@ class ConversationList extends React.PureComponent<Props> {
   // Was using onEndReached but that was really flakey
   _onViewableItemsChanged = ({viewableItems}) => {
     const topRecord = viewableItems[viewableItems.length - 1]
+    const bottomRecord = viewableItems[0]
     if (topRecord && topRecord.item === 'specialTop') {
       const ordinalRecord = viewableItems[viewableItems.length - 2]
       // ignore if we don't have real messages
       if (ordinalRecord && ordinalRecord.item !== 'specialBottom') {
         this.props.loadOlderMessages(this.props.messageOrdinals.get(ordinalRecord.item))
       }
     }
+    if (!topRecord || !bottomRecord) {
+      _debug(`_onViewableItemsChanged: bailing out because of no record`)
+      return
+    }
+
+    const bottomIndex = this._getIndexFromItem(viewableItems[0].item)
+    const upperIndex = this._getIndexFromItem(viewableItems[viewableItems.length - 1].item)
+    const middleIndex = bottomIndex + Math.floor((upperIndex - bottomIndex) / 2)
+    _debug(`_onViewableItemsChanged: first: ${bottomIndex} last: ${upperIndex} middle: ${middleIndex}`)
+    if (!this._scrollCenterTarget) {
+      _debug(`_onViewableItemsChanged: no center target`)
+      return
+    }
+
+    if (
+      !(
+        this._scrollCenterTarget <= middleIndex + targetHitArea &&
+        this._scrollCenterTarget >= middleIndex - targetHitArea
+      )
+    ) {
+      _debug(`_onViewableItemsChanged: scrolling to: ${this._scrollCenterTarget}`)
+      this._scrollToCentered()
+    } else {
+      _debug(`_onViewableItemsChanged: cleared`)
+      this._scrollCenterTarget = null
+    }
   }
 
   // not highly documented. keeps new content from shifting around the list if you're scrolled up
   _maintainVisibleContentPosition = {
     minIndexForVisible: 0,
   }
 
+  _jumpToRecent = () => {
+    const list = this._listRef.current
+    if (list) {
+      const index = this._getOrdinalIndex(this.props.messageOrdinals.last())
```

Comment:
could this just be `this.props.messageOrdinals.size`? (or maybe 0 because of the inverted list)

Reply:
Yeah I think so, I was going to change it, but thought maybe this was more general (since this list has these special items in it "specialTop" and "specialBottom", I was wary of just picking index 0 or 1)